packages/design-system/src/components/Modal/Modal.stories.tsx

```diff
@@ -1,39 +1,249 @@
-import React from 'react';
+import React, { ReactElement, useState } from 'react';
+import { ComponentStory } from '@storybook/react';
 import { action } from '@storybook/addon-actions';
 
-import { ButtonPrimary, ButtonSecondary, ButtonDestructive } from '../Button';
-
-import Modal from '.';
+import { ButtonPrimary } from '../Button';
+import Modal, { ModalPropsType } from './Modal';
 
 export default {
 	component: Modal,
 };
 
-export const Confirm = () => (
-	<Modal.Body data-enter>
-		<Modal.Heading>
-			<h1>Leave page</h1>
-		</Modal.Heading>
-		<p>Your change will be lost. Do you want to leave the page without saving?</p>
-		<Modal.Buttons>
-			<ButtonSecondary onClick={action('Clicked continue')}>Continue editing</ButtonSecondary>
-			<ButtonDestructive onClick={action('Clicked discard')}>Discard changes</ButtonDestructive>
-		</Modal.Buttons>
-	</Modal.Body>
-);
-Confirm.parameters = {};
-
-export const Invite = () => (
+function ModalStory(props: Partial<ModalPropsType>) {
+	const [modalOpen, setModalOpen] = useState(false);
+
+	return (
+		<>
+			<button onClick={() => setModalOpen(true)} data-test="open-modal">
+				See
+			</button>
+
+			{modalOpen && (
+				<Modal
+					header={{ title: '(Default story title)' }}
+					children="(Default story child)"
+					onClose={() => {
+						action('onClose');
+						setModalOpen(false);
+					}}
+					{...props}
+				/>
+			)}
+		</>
+	);
+}
+
+function QuicheRecipe(): ReactElement {
```

Comment:
```suggestion
function QuicheRecipe() {
```
Do we really need to set the type?

Reply:
Not really needed (TS inference would probably work here)